### PR TITLE
[fix](schema change) fix the defineName field is not the same when copying column

### PR DESCRIPTION
**Pull request by luwei16**
The copy constructor of the column class calls the getDefineName function, which causes the dest column defineName to have a specific value when src column defineName is null. The value of defineName is different after copying. So call getRealDefineName instead

```
public String getDefineName() {
    if (defineName != null) {
        return defineName;
    }
    return name;
}
```

```
public String getRealDefineName() {
    return defineName;
}
```

### CODE REVIEW

**Comment by luwei16** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java [hunk 1]
```diff
@@ -341,6 +341,10 @@ public String getDefineName() {
         return name;
     }
 
+    public String getRealDefineName() {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java:344]: add comment

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: regression-test/suites/schema_change_p0/test_add_rename_column.groovy [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    sql """ ALTER TABLE ${tableName} add column c3 varchar(10) """
+
+    sql """ ALTER TABLE ${tableName} rename column c3 c4 """
+
+    sql """ truncate table ${tableName} """
+
+    qt_sql """ select * from ${tableName} """
+
+    sql """ insert into ${tableName} values (3, 'cc', 'dd') """
+
```
Comment (yiguolei) [inline regression-test/suites/schema_change_p0/test_add_rename_column.groovy:50]: add sync,
In multi FE env, the result will not stable if not sync

Comment (luwei16) [reply regression-test/suites/schema_change_p0/test_add_rename_column.groovy:52]: done

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java (hunk 1 above)
Comment (luwei16) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java:346]: > add comment

done

**Comment by luwei16** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**